backend/services/itemService.js

```diff
@@ -55,9 +62,43 @@ module.exports = {
         }
     },
 
-    update: async (params) => {
+    update: async (before, to) => {
         try {
+
+            // Before's status change
+            await Item.findOneAndUpdate({ _id: before._id }, { status: 'modified' });
+
+            // Append history
+            to = Object.assign(to, { history: [...before.history, before._id]});
+
+            // Copy Item
+            let item = Object.assign(before, to).toJSON();
+            delete item._id
+
+            // console.log(item.contributors[0], item.contributors[0].toString)
+
+            // Making unique array
+            item.contributors = [...new Set(item.contributors.map(obj => obj.toString()))]
```

Comment:
Additional explanation of `[... new Set(...)]` syntax means would be nice. I found it hard to intuitively understand what is going on here.
You can add links to a stackoverflow answer or a article online to help readers understand the logic behind it.